### PR TITLE
Rework CI, stop synchronizing with VSCode versions

### LINKED ISSUE #291: Proposal: Use a different versioning scheme
The current situation is: If you upgrade from `1.83.12` to `1.83.16` it seems like a patch, but it really isn't. 😅   If I use `~1.83.12` as a version selector in a dependent project you expect non-breaking changes. The only way to prevent this right now is to pin-point it to a fixed version like this: `1.83.12`.

What do you think about the following version scheme?
185.0.0 = [vscode-version].[mva-level-changes].[mva-bugfixes]

So, new features, breaking changes will be 185.1.0, and bug fixes 185.0.1 like in regular semantic versioning. The meaning of minor is a bit stretched here, but there is at least the possibility to distinguish.

### PR DESCRIPTION
**Pull request by CGNonofr**
based on https://github.com/CodinGame/monaco-vscode-api/pull/327

fix #291

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -93,10 +81,10 @@ See [this issue](https://github.com/CodinGame/monaco-vscode-api/issues/186) or t
 
 ## Monaco standalone services
 
-Also, monaco-editor use `standalone` versions or the vscode services, which are much simpler.
+`monaco-editor`, as well as this library by default, uses `standalone` versions or the vscode services, which are much simpler than the one used in VSCode.
```
Comment (dkattan) [inline README.md:84]: standalone version _of_ vscode services

File: README.md [hunk 2]
```diff
@@ -239,11 +227,20 @@ updateUserConfiguration(`{
...
-The editors created using `monaco.editor.create` don't use the configuration from the configurationService.
+The recommended way is to used the `createModelReference` method instead (added on top of the official monaco-editor api) which returns instead a reference to a model.
 
-This library exposes functions to create editors binded on the configuration service:
+It has some pros:
+- The model reference can be used by VSCode services, allowing for instance following links between files (ctrl+click)
+- The returned model is bound to a filesystem file, and you have access to methods allowing to control the file lifecycle (saving the file, accessing the dirty state...)
+- It is possible to call the method multiple times on the same file to get multiple references. The model is disposed when there is no reference left
+
+The second argument of the method allows to write the file content to the virtual filesystem in case the file wasn't registered in it beforehand.
```
Comment (dkattan) [inline README.md:243]: The second argument of the method allows _you_ to write the

**Comment by dkattan** [comment]
This is very exciting. I've always felt like the treemending approach particularly the postinstall hook was very brittle and this appears to clean a lot of that up. Love the work you are doing here!

**Comment by CGNonofr** [comment]
> This is very exciting. I've always felt like the treemending approach particularly the postinstall hook was very brittle and this appears to clean a lot of that up. Love the work you are doing here!

Thank you for your kind words and you feedback :)

**Comment by CGNonofr** [comment]
> I've always felt like the treemending approach particularly the postinstall hook was very brittle and this appears to clean a lot of that up.

Me too!

btw, there was already an alternative to the postinstall hook: the monaco-editor-treemended package

**Comment by CGNonofr** [comment]
@kaisalmen we need this PR to release a v2, otherwise it will just release a 1.85.7

**Comment by kaisalmen** [comment]
@CGNonofr is it ready? It is a Draft currently. 😉

**Comment by CGNonofr** [comment]
> @CGNonofr is it ready? It is a Draft currently. 😉

my bad :laughing: it's ready now (:

**Review by kaisalmen — APPROVED** [review]
I can't really judge if this is working or not as I don't have the means to fully test it. Please go ahead. The released artifacts are working as expected, thought.

**Comment by CGNonofr** [comment]
> I can't really judge if this is working or not as I don't have the means to fully test it. Please go ahead. The released artifacts are working as expected, thought.

Note that we can now release a next version using the github CI directly, and I've used it to release the version you're testing. The differences between the main and next channels are handled by semantic-release itself so it's probably quite safe

**Comment by kaisalmen** [comment]
There is a problem. v2.0.0 does not carry the latest tag:

**Comment by CGNonofr** [comment]
> @CGNonofr There is a problem. v2.0.0 does not carry the latest tag: image

Oh no :cry:

**Pull request merged**